### ISSUE TITLE
[BUG]Can't add secondary disk as data volumen at 2nd host of cluster

### ISSUE THREAD

**Issue by insbire** [harvester/harvester#1755]
**Describe the bug**

3 Baremetal Server with same hardware cofiguration.
sda: SSD driver, installed Harvester 1.0
sdb: SATA driver, should be added as data volumen to host

**To Reproduce**
1. At Master Node, all works as expected.
![Harvester 1 0 Cluster Master Node](https://user-images.githubusercontent.com/86938244/147553673-a865dcfd-284e-48f5-99dd-6edc8a8daf3b.png)
2. At the 2nd Node, the data disk can be detected, but couldn't be added as data volumen to the host.
![Harvester 1 0 Cluster Secodary Node](https://user-images.githubusercontent.com/86938244/147553843-75a7b7c5-3c51-4605-b0f8-947c4ed5f69c.png)
3. The 3rd Node has same problem.
![image](https://user-images.githubusercontent.com/86938244/147555324-4a809db3-2b84-480c-80d2-2b32240aa575.png)

**Expected behavior**

The secondary disk should be added as data volumen at 2nd and 3rd host as well.
**Support bundle**

[supportbundle_b5a94f11-01e1-4181-ad74-b141d4d82809_2021-12-28T10-11-12Z.zip](https://github.com/harvester/harvester/files/7784112/supportbundle_b5a94f11-01e1-4181-ad74-b141d4d82809_2021-12-28T10-11-12Z.zip)

**Environment:**
 - Harvester ISO version: 1.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal Dell R740

**Additional context**
All sdbs were formatted as ext4,  mkfs -t ext4 /dev/sdb

**Comment by Palando** [comment]
I have the same problem with three Intel NUCs. On the first node I could add the second disc (sdb) as a data volume. On the 2nd and 3rd node it can not be added although it is detected. All nodes are the same hardware.

**Comment by insbire** [comment]
I tried the second time to install Harvester 1.0 on the 3 nodes of cluster. In order to figure it out, the reason which causes the issue, did the following behavior:
1. Install Harvester 1.0 on the 3 nodes.
![01_reinstall_harvester](https://user-images.githubusercontent.com/86938244/148035553-86ab191c-a7b3-4c04-befd-8e27aa33a0c7.png)

2. Before add sdb on each node, check the availability of sdbs.
![02 status alpha - 2 sdbs](https://user-images.githubusercontent.com/86938244/148035906-7b682fff-2ec9-4662-9d35-3059f5319deb.png)
1st node: disk-> /dev/sdb; partition-> /dev/sdb1
![02 status bravo - sdb-sda](https://user-images.githubusercontent.com/86938244/148035937-05578060-9ac7-4418-9bd1-4cb34c1ec38d.png)
2nd node: disk-> /dev/sda  😮
![02 status charlie - sdb-sda](https://user-images.githubusercontent.com/86938244/148035954-6746ea55-dc36-44c5-af4e-e431fb6ffb20.png)
3rd node: disk-> /dev/sda,  same as 2nd node.

3. Get ready to add sdb on 1st node, but ...
![03 alpha disks](https://user-images.githubusercontent.com/86938244/148038411-f35efaa0-6a31-4106-858e-f321df685322.png)
I got 2 sdbs. 😱

4. Check device information on 3 nodes.
![03 alpha lsblk](https://user-images.githubusercontent.com/86938244/148038865-5ab4df53-0aa4-4584-8391-04898b34dd2c.png)

![03 bravo lsblk](https://user-images.githubusercontent.com/86938244/148038985-58d23628-b38f-4198-abfa-2991ee8bd177.png)

![03 charlie lsblk](https://user-images.githubusercontent.com/86938244/148039147-11214365-f31c-4f7a-9b62-ef58dfaa66da.png)

After Installation, the SSD disk and hard disk swapped their name.

5. On 2nd node, I did quickly enough, to add **partition** /dev/sda1 on the node.
![04 bravo disks](https://user-images.githubusercontent.com/86938244/148042044-3df8f21f-4ab1-4924-88d7-6136f4733502.png)

6. On 3rd node, got the same result as the first installation.
![04 charlie disks](https://user-images.githubusercontent.com/86938244/148042146-43f9cbfc-f29a-48a7-aece-45c684d4b5d6.png)

7. On the 1st node, added the wrong partition /dev/sdb1, couldn't remove it.
![05 partition disk conflict](https://user-images.githubusercontent.com/86938244/148043003-8b11e9e1-b1a3-43e1-bcde-a5df3ade66d4.png)

My conclusion:
It seems like, the issue is consequence by the swapped name of sda and sdb. 
On our 4th server, which has only sda, has no problem at all.
![03 nf5280m5 lsblk](https://user-images.githubusercontent.com/86938244/148043707-253657d2-5b33-4e86-846a-97e87ffbf6cb.png)
![image](https://user-images.githubusercontent.com/86938244/148044888-5bf62714-7ba4-4ac6-8ad1-636a286d17a8.png)

So, this issue could be caused only on the node with more than 2 disk devices.  Obviously！😂

**Comment by ebauman** [comment]
Also seeing this issue. 3 node harvester cluster, first node I could add all of the disks. 2nd and 3rd, no dice.

**Comment by weihanglo** [comment]
Not familiar with Vue but it seems that the getter method `childParts` doesn't exclude blockdevice from other nodes.

https://github.com/harvester/dashboard/blob/v1.0.0/models/harvester/harvesterhci.io.blockdevice.js#L5-L12

On the other hand, the blockdevices on host resource does filter by `blockdevice.spec.nodeName`.

https://github.com/harvester/dashboard/blob/v1.0.0/edit/harvesterhci.io.host/index.vue#L71-L78

@n313893254, can you help verify if this is the root cause?

**Comment by n313893254** [comment]
@weihanglo Thanks, It is the root cause.

**Reproduce Steps**
- Install Harvester with two nodes
- Add a disk(USB) for each node
- For the first node, add the part to the node.
- Go to the second node, check "Add disk"

**Result**
The disk can not be added as data volume to the host.

**Comment by TachunLin** [comment]
Verified **fixed** on master-1be9cb5e-head (1/23). Close this issue.

### Result
1. The secondary disk can be added as data volume at 2nd and 3rd host after adding the 1st node 
  - Node 1, 2, 3 disk status before adding second disk
    ![image](https://user-images.githubusercontent.com/29251855/150683305-a78e06fb-06bd-4403-b518-388bdb87e9b2.png)
  - Node 1, 2, 3 disk status after adding second disk
    ![image](https://user-images.githubusercontent.com/29251855/150683997-bcb5ea79-f005-4971-a587-4355c51e91f1.png)
    
2. Can correctly add second disk `/dev/sdb` to `node 2` 
![image](https://user-images.githubusercontent.com/29251855/150683942-e97ef2a5-ae38-49d3-9109-e408e33b696f.png)
 
3. Can correctly add second disk to `/dev/sdb` to `node 3` 
![image](https://user-images.githubusercontent.com/29251855/150684191-03880121-adc2-4060-a10b-bd02154d045d.png)

4. Can correctly add second disk `/dev/sdb` with `no gpt partition` to `node 1` 
![image](https://user-images.githubusercontent.com/29251855/150683902-8bf85f24-252f-4d0e-b8cf-9cbeca96dba7.png)

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-1be9cb5e-head (1/23)
* Disk status 
  - node 1: 2nd disk /dev/sdb (no gpt partition)
  - node 2: 2nd disk /dev/sdb (with gpt partition)
  - node 3: 2nd disk /dev/sdb (with gpt partition) 

### Verify Steps

1. Prepare a 3 nodes harvester each have a second disk `/dev/sdb` ready to be added to host 
2. Add the second disk on node 1 disk page in host  
3. Edit disk page on node 2 
4. Check the /dev/sdb disk is available 
5. Add the second disk on node 2 disk page in host 
6. Edit disk page on node 3
7. Check the /dev/sdb disk is available 
8. Add the second disk on node 2 disk page in host 

### Additional Context
If the second disk status is not display in `scheduled` status, please press `ctrl + r` to refresh UI

Below is details check 
* Node 1, 2, 3 disk status before adding second disk
![image](https://user-images.githubusercontent.com/29251855/150683305-a78e06fb-06bd-4403-b518-388bdb87e9b2.png)


* Node 1 before adding second disk 
  ![image](https://user-images.githubusercontent.com/29251855/150681624-2376bc2e-f48d-4bb1-83a9-430f74b9a463.png)

* Node 1 disk status 
  ```
  rancher@hpd8s7:~> lsblk
  NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
  loop0    7:0    0   3.2G  1 loop /
  sda      8:0    0 279.4G  0 disk 
  ├─sda1   8:1    0    47M  0 part 
  ├─sda2   8:2    0    50M  0 part /oem
  ├─sda3   8:3    0    15G  0 part /run/initramfs/cos-state
  ├─sda4   8:4    0     8G  0 part 
  ├─sda5   8:5    0    60G  0 part /usr/local
  └─sda6   8:6    0 196.3G  0 part /var/lib/longhorn
  sdb      8:16   0 558.7G  0 disk 
  sdc      8:32   0    50G  0 disk /var/lib/kubelet/pods/6e484470-474b-4c3b-a98a-15c71192155a/volume-subpaths/pvc-3344c591-0ba3-41ba-ba2c-591ab543
  sdd      8:48   0    10M  0 disk /var/lib/kubelet/pods/745127aa-d3aa-43c7-9c7e-d27ec4dcf043/volumes/kubernetes.io~csi/pvc-1298033b-a494-4efd-bd9
  ```

* Node1 `/dev/sdb` have **no gpt partition**
  ```
  hpd8s7:~ # gdisk /dev/sdb
  GPT fdisk (gdisk) version 1.0.1
  
  Partition table scan:
    MBR: not present
    BSD: not present
    APM: not present
    GPT: not present
  
  Creating new GPT entries.
  ```

* Node 2 before adding second disk
  ![image](https://user-images.githubusercontent.com/29251855/150681648-42dc03be-b833-458f-9bd8-7c8febe0efec.png)

* Node 2 disk status  
  ```
  rancher@harvester-qg7wt:~> lsblk
  NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
  loop0    7:0    0   3.2G  1 loop /
  sda      8:0    0 279.4G  0 disk 
  ├─sda1   8:1    0    47M  0 part 
  ├─sda2   8:2    0    50M  0 part /oem
  ├─sda3   8:3    0    15G  0 part /run/initramfs/cos-state
  ├─sda4   8:4    0     8G  0 part 
  ├─sda5   8:5    0    60G  0 part /usr/local
  └─sda6   8:6    0 196.3G  0 part /var/lib/longhorn
  sdb      8:16   0 558.7G  0 disk 
  └─sdb1   8:17   0 558.7G  0 part
  ```

* Node2 `/dev/sdb` have **gpt partition**
  ```
  harvester-qg7wt:~ # gdisk /dev/sdb
  GPT fdisk (gdisk) version 1.0.1
  
  Partition table scan:
    MBR: protective
    BSD: not present
    APM: not present
    GPT: present
  
  Found valid GPT with protective MBR; using GPT.
  ```


* Node 3 before adding disk
![image](https://user-images.githubusercontent.com/29251855/150682015-9625fc94-2f8d-4e7f-87a8-b1d5a16fe775.png)


* Node 3 disk status
  ```
  rancher@harvester-d7fkx:~> lsblk
  NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
  loop0    7:0    0   3.2G  1 loop /
  sda      8:0    0 279.4G  0 disk 
  ├─sda1   8:1    0    47M  0 part 
  ├─sda2   8:2    0    50M  0 part /oem
  ├─sda3   8:3    0    15G  0 part /run/initramfs/cos-state
  ├─sda4   8:4    0     8G  0 part 
  ├─sda5   8:5    0    60G  0 part /usr/local
  └─sda6   8:6    0 196.3G  0 part /var/lib/longhorn
  sdb      8:16   0 558.7G  0 disk 
  └─sdb1   8:17   0 558.7G  0 part
  ```

* Node3 `/dev/sdb` have **gpt partition**
  ```
  harvester-d7fkx:~ # gdisk /dev/sdb
  GPT fdisk (gdisk) version 1.0.1
  
  Partition table scan:
    MBR: protective
    BSD: not present
    APM: not present
    GPT: present
  
  Found valid GPT with protective MBR; using GPT.
  ```

* Add second disk on node 1
![image](https://user-images.githubusercontent.com/29251855/150683693-26cf1a3b-afd4-4ff2-927d-ac47d47681b4.png)

* Unable to add disk without gpt partition
![image](https://user-images.githubusercontent.com/29251855/150683714-03a2cc6a-9022-45f9-9e7f-716bc34b2a74.png)

* Node 1 disk status with second disk
![image](https://user-images.githubusercontent.com/29251855/150683902-8bf85f24-252f-4d0e-b8cf-9cbeca96dba7.png)

* Node 2 disk status with second disk
![image](https://user-images.githubusercontent.com/29251855/150683942-e97ef2a5-ae38-49d3-9109-e408e33b696f.png)

* Node 3 disk status with second disk
![image](https://user-images.githubusercontent.com/29251855/150684191-03880121-adc2-4060-a10b-bd02154d045d.png)
 

* Node 1, 2, 3 disk status after adding second disk
![image](https://user-images.githubusercontent.com/29251855/150683997-bcb5ea79-f005-4971-a587-4355c51e91f1.png)
